nearuplib/nodelib.py

```diff
@@ -53,6 +53,31 @@ def get_chain_id_from_flags(flags):
     return ''
 
 
+def genesis_changed(chain_id, home_dir):
+    genesis_md5sum = get_genesis_md5sum(chain_id)
+    local_genesis_md5sum = hashlib.md5(open(os.path.join(
+        os.path.join(home_dir, 'genesis.json')), 'rb').read()).hexdigest()
+    if genesis_md5sum == local_genesis_md5sum:
+        print(f'Our genesis version up to date')
+        return False
+    else:
+        print(
+            f'Remote genesis protocol version md5 {genesis_md5sum}, ours is {local_genesis_md5sum}')
+        return True
+
+
+def check_and_update_genesis(chain_id, home_dir):
+    if genesis_changed(chain_id, home_dir):
+        print(
+            f'Update genesis config and remove stale data for {chain_id}')
+        os.remove(os.path.join(home_dir, 'genesis.json'))
+        download_genesis(chain_id, home_dir)
+        if os.path.exists(os.path.join(home_dir, 'data')):
+            shutil.rmtree(os.path.join(home_dir, 'data'))
```

Comment:
I'm wondering if this is the only way to handle genesis changes rn? This is suboptimal when state is very large and we create genesis file from that state. Even nodes that are up to date will require very large time to download new genesis file. 

Notice that this is a generic concern outside of the scope of `nearup`. This should be handled as part of the upgradability mechanism.

Reply:
Well if genesis changed how do you expect old data to still work?

Reply:
Yeah when genesis change, our validator node also start from empty db. As of now new genesis file is not very large size, about 200M for current testnet. We've discussed to migrate state not in genesis, but in storage, if that happen, some of our current genesis-changing protocol change will be a storage migration on old data, saving a lot of download